### PR TITLE
Add Reviews to Show pages

### PR DESCRIPTION
**Pull request by BenWoodford76**
Adding relevant code for displaying show reviews on the relevant pages.
Also adds data for all 2015-16 season shows and Autumn 2016 semester shows

### CODE REVIEW

**Comment by BenWoodford76** [comment]
This has gone astoundingly well.........

**Comment by nhyty** [comment]
@wjdp , can you rebase this onto master to fix half the PR build fail

File: _shows/15_16/bull.md [hunk 1]
```diff
@@ -37,6 +37,26 @@ crew:
...
+  - type: Review
+    href: www.thereviewshub.com/bull-nottingham-new-theatre
+    snapshot: rHjlZ
+    publisher: TheReviewsHub 
+    author: Phil Lowe
+    title: "Bull - Nottingham New Theatre"
+    date: 2016-02-25
+    stars: 4
+    quote: "This is one play in which the good guy doesn't win in the end although it is questionable who in the audience would like to see him do so[...] this sterling production by Nottingham New Theatre starts their new season off with a tremendous bang. "
+  - type: Review
```
Comment (nhyty) [inline _shows/15_16/bull.md]: Issue is with this line

**Comment by BenWoodford76** [comment]
@nhyty I'm assuming that'll fix the conflicting files? @nhyty

File: _shows/15_16/electric_nebraska.md [hunk 2]
```diff
@@ -41,6 +41,17 @@ crew:
    name: Harry Bradley
 
 prod_shots: 5t3Lrx
+
+links:
+    - type: Review
```
Comment (nhyty) [inline _shows/15_16/electric_nebraska.md]: Indenting issue here!

File: _shows/15_16/aaron_calvert.md [hunk 3]
```diff
@@ -10,7 +10,16 @@ assets:
   - type: poster
     image: k6zSXmF
 
-
+links:
+- type: Review
```
Comment (nhyty) [inline _shows/15_16/aaron_calvert.md]: Indenting issues here

**Comment by wjdp** [comment]
Will have a look at this tomorrow, probably.

**Comment by samozzy** [comment]
In terms of the actual reviews, they need to be styled better before this gets merged.
Also want to new tab the full review links.
Talking of - how do we determine which link to use, archived snapshot or live link?

**Comment by wjdp** [comment]
@samozzy Leave snapshots out of the layout for the moment. We'll address this later.

**Comment by wjdp** [comment]
```
years/15_16/constellations/index.html
  target does not exist --- years/15_16/constellations/index.html --> www.impactnottingham.com/2016/06/constellations-nottingham-new-theatre/
  target does not exist --- years/15_16/constellations/index.html --> www.impactnottingham.com/2016/06/constellations-nottingham-new-theatre/
years/15_16/dark_room/index.html
  target does not exist --- years/15_16/dark_room/index.html --> www.hercampus.com/school/nottingham/review-dark-room-nottingham-new-theatre
  target does not exist --- years/15_16/dark_room/index.html --> www.hercampus.com/school/nottingham/review-dark-room-nottingham-new-theatre
  target does not exist --- years/15_16/dark_room/index.html --> www.impactnottingham.com/2015/11/dark-room-nottingham-new-theatre
  target does not exist --- years/15_16/dark_room/index.html --> www.impactnottingham.com/2015/11/dark-room-nottingham-new-theatre
```

External URLs need `http://` or `https://` in front of them

**Comment by BenWoodford76** [comment]
@wjdp I'll go through and sort this when I get a free afternoon

**Comment by BenWoodford76** [comment]
@samozzy I'm happy to add the data if you want to concentrate on the code and styling?

**Comment by BenWoodford76** [comment]
Also, on the when to use live link vs. Snapshot question, maybe use the live link until the end of that academic year, then switch to the snapshot? @wjdp @samozzy thoughts?

**Comment by wjdp** [comment]
Having looked into this a bit re. link template please see `_includes/link-list.html`. This is the list already in use by the people pages and the template built here should be integrated with it. There is already defined behaviour about snapshots vs real URLs there—which may need refinement—but that's not really in scope here.

**Comment by BenWoodford76** [comment]
Am I leaving this branch to be merged now or can I add more reviews to it?

**Comment by wjdp** [comment]
No more extra stuff in here, much bloated already. Just spin up another branch and PR.

**Comment by BenWoodford76** [comment]
@wjdp Will do

**Comment by samozzy** [comment]
@wjdp Assume you're happy for me to add extra things in the form of styling over the weekend still though?

**Comment by wjdp** [comment]
@samozzy Yep, that's the stuff we need to get the PR accepted. Adding other items (more reviews) could add more things to go wrong/fix which would slow down getting it merged.

File: _includes/link-list.html [hunk 4]
```diff
@@ -1,14 +1,52 @@
 <dl class="link-list">
-{% for link in include.list %}
+{% for link in include.links %}
+
+{% if link.href_snapshot %}
```
Comment (wjdp) [inline _includes/link-list.html]: Selection of the link href is done in [LinkList](https://github.com/newtheatre/history-project/blob/master/_plugins/link_list.rb).

Apologies, this API isn't really documented (at all), but roughly sits inbetween the data in the show file (or other) and tweaks it before the this template gets ahold of it. This block should be removed and we just use `link.href` to refer to the resource.

File: _includes/link-list.html [hunk 5]
```diff
@@ -1,14 +1,52 @@
...
+		{% endfor %}
+
+		<strong>{{ link.publisher }}</strong> - <em>"{{ link.quote }}" - <a href="{{ linkURL }}">Full Review</a></em> 
+		<a href="{{ linkURL }}" target="_blank"><small><i class="fa fa-external-link"></i></small></a>
+
 	{% else %}
-	  <a href="{{ link.href }}" {{ link.data }}>{{ link.title }}</a>
-	{% endif %}
+
+  <a href="{{ linkURL }}" {{ link.data }}>{{ link.title }}</a>
```
Comment (wjdp) [inline _includes/link-list.html]: Can we remove changes to these lines, looks like extra whitespace and removal of indents.

**Comment by nhyty** [comment]
How close is this to be merged in???

**Comment by samozzy** [comment]
@nhyty Once I've done Will's reviews, which will probably be Monday when I'm at home.

**Comment by nhyty** [comment]
Any news on this @samozzy

**Comment by samozzy** [comment]
@nhyty Job for this evening :)

**Comment by BenWoodford76** [comment]
Any updates? :)

**Comment by samozzy** [comment]
@BenWoodford76 Think this is good to go now.

**Comment by nhyty** [comment]
@wjdp, hada chance to look over this yet??

**Comment by wjdp** [comment]
Yep, I've made a number of changes re. style and a fix where people links were broken.

Reviews now look like:

image

I've taken out the opening of links in a `_blank`, out of scope for this PR and introduces touch issues.

Can I get a review or two, if approved I'll merge this evening.

**Comment by BenWoodford76** [comment]
@wjdp Is it converting the 'out of ten' ratings from impact into a star rating??

**Comment by wjdp** [comment]
@BenWoodford76 It is, the view from a bridge reviews are `rating:x/10` but the bottom one should be 3/5 (source is 6/10). Odd

**Comment by nhyty** [comment]
One note i have is the consistancy of names of reviewing groups, for example both impact and hercampus have at least two different names throughout just these.

**Comment by BenWoodford76** [comment]
@nhyty Im happy to go through them all once merged and sort consistency of names

**Comment by samozzy** [comment]
Looks lovely - thank you. Yes, the scaled from out of ten to our out of five is my logic. Anything not out of five should be given as a rating and is then subject to scaling. 

For bikeshedding purposes: I think HerCampus should have Nottingham, as they also do other universities within the same website.

**Comment by BenWoodford76** [comment]
@samozzy Agree with the bikeshedding

**Comment by nhyty** [comment]
@samozzy I agree, for 'franchise' media outlets i feel location is important. in case we ever take anything to another uni, which is a goal next year apparently.

**Comment by nhyty** [comment]
happy to merge this in now @wjdp ???

**Pull request merged**